wandb/apis/workspaces/interface.py

```diff
@@ -0,0 +1,449 @@
+        )
+
+    def to_model(self):
+        if (section_settings := self.section_panel_settings) is None:
+            section_settings = SectionPanelSettings()
+
+        section_settings_model = section_settings.to_model()
+        panel_models = [p.to_model() for p in self.panels]
+
+        # Add warning that panel layout only works if they set section settings layout = "custom"
+
+        return internal.PanelBankConfigSectionsItem(
+            name=self.name,
+            panels=panel_models,
+            is_open=not self.collapsed,
+            flow_config=section_settings_model,
+        )
+
+
+@dataclass(config=dataclass_config, repr=False)
+class WorkspaceSettings(Base):
+    """Settings for the workspace, typically seen at the top of the workspace in the UI.
+
+    This object includes settings for the x-axis, smoothing, outliers, panels, tooltips, runs, and panel query bar.
+
+    Settings applied here can be overrided by more granular Section and Panel settings.
+
+    Args:
+        x_axis (str): Global x-axis setting.
+        x_min (float): The minimum value for the x-axis.
+        x_max (float): The maximum value for the x-axis.
+        smoothing_type (SmoothingType): The type of smoothing to apply to the data.
+        smoothing_weight (float): The weighting factor for smoothing.
+        ignore_outliers (bool): Whether to ignore outliers in charts
+        remove_legends_from_panels (bool): Whether legends should be removed from panels.
+        tooltip_number_of_runs (str): The number of runs to show in the tooltip.
+        tooltip_color_run_names (bool): Whether to color run names in the tooltip.
+        max_runs (int): The maximum number of runs to display.
+        point_visualization_method (str): Controls sampling method for points
+        panel_search_query (str): The query for the panel search bar.
+        auto_expand_panel_search_results (bool): Whether to auto expand panel search results.
+    """
+
+    # Axis settings
+    x_axis: str = "_step"  # fix this to use name map in future
+    x_min: Optional[Annotated[float, Ge(0)]] = None
+    x_max: Optional[Annotated[float, Ge(0)]] = None
+
+    # Smoothing settings
+    smoothing_type: internal.SmoothingType = "none"
+    smoothing_weight: Annotated[float, Ge(0)] = 0
+
+    # Outlier settings
+    ignore_outliers: bool = False
+
+    # Panel settings
+    remove_legends_from_panels: bool = False
+
+    # Tooltip settings
+    tooltip_number_of_runs: Literal["single", "default", "all_runs"] = "default"
+    tooltip_color_run_names: bool = True
+
+    # Run settings
+    max_runs: PositiveInt = 10
+    point_visualization_method: Literal["bucketing", "downsampling"] = "bucketing"
+
+    # Panel query bar settings
+    panel_search_query: str = ""
+    auto_expand_panel_search_results: bool = False
+    _panel_search_history: Optional[LList[Dict[Literal["query"], str]]] = Field(
+        None, init=False, repr=False
+    )
```

Comment:
Yeah it's unfortunate because those type defs don't reflect what the user sees in the UI today -- there may be some default values that get added later.

For example, if you create a new workspace, the `maxRuns` field is always 10, even though the defaults on the type don't say that.

The intent of this API is to match what the user sees in the FE, so we use whatever defaults the UI has.